### ISSUE TITLE
Need the Slicing Tolerance option from Curas experimental settings

### ISSUE THREAD

**Issue by tjc2005** [SoftFever/OrcaSlicer#1186]
Part tolerance is not good, I have multiple objects that won't fit together, however if I use cura on any of my printers and use exclusive mode in cura experimental settings, things work much better. With the same print settings.

Explanation here https://support.makerbot.com/s/article/1667418054188

The only way to get it to work for me currently is to make a part smaller or bigger but then that presents it's own set of issues with tolerances.

Can we get these added to orcaslicer?

**Comment by rhiffsta** [comment]
i second this. a really useful option that is missing in orca slicer. please add these options.

**Comment by PowerWiesel** [comment]
I would love to see that feature in Orca.
That's one of the reasons why I fallback to Cura sometimes

**Comment by dbuezas** [comment]
Isn't this practically solved by x-y hole and contour compensation?

**Comment by tjc2005** [comment]
> Isn't this practically solved by x-y hole and contour compensation?

No it's not the same unfortunately.

**Comment by tjc2005** [comment]
> I mean, it used to fit only the .3mm tolerance, with x-y hole compensation set to 0.15mm it is spot on. <img alt="image" width="811" src="https://private-user-images.githubusercontent.com/777196/287542993-53977084-9a7a-4c1d-99e1-21493a3d297d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2MzM1MzAsIm5iZiI6MTcwMTYzMzIzMCwicGF0aCI6Ii83NzcxOTYvMjg3NTQyOTkzLTUzOTc3MDg0LTlhN2EtNGMxZC05OWUxLTIxNDkzYTNkMjk3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwM1QxOTUzNTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMWYzZDJjNDJiNDgyODlhOTIwMzViZTRjZWVlMTA1MjQ3MDdjY2NiMGRlYzcxNzZjZGI0ZTVkMWJjYTFmMjk0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.e-tJYnRi5KM8Jqubgw1mwTNqAJEPRR80aTJV77nPEno">
> 
> (although I agree that Cura's tolerance seems less of a h
![IMG_20231203_195514](https://github.com/SoftFever/OrcaSlicer/assets/25927795/4b0ebe9c-ab12-45b6-949f-6ba2ca522b6a)
ack)

**Comment by dbuezas** [comment]
I mean xy-hole (and xy contour) compensation, not xy-size compensation

**Comment by dbuezas** [comment]
I'd suggest you do this:
1. print an orca tolerance test
2. see where the printed pin fits as tight as you wish tolerances to be
3. Divide the number by 2
4. Set it as xy hole compensation
5. Print the tolerance test again and corroborate it now fits in the zero position but it is not too loose

**Comment by PowerWiesel** [comment]
Did you try this with threads? Just print a simple screw with idk 5mm OD and a nut and test if it works smoother.
It's probably also an issue with the tolerance of my printer, but on on my last design I wasn't able to get it screwed in more then one turn.
After I resliced and reprinted just one of both parts with that Cura setting it worked with almost no resistance. The difference is unbelievable.
The thread was also pretty big - 80mm OD. That might have been a factor.

**Comment by markdid** [comment]
Was just looking for this feature

**Comment by devinSpitz** [comment]
That is true for me as well as I print this design: https://www.printables.com/de/model/510256

Does work with Cura and "Slicing Tolerance" to "Exclusive" but not with the Orca slicer default settings.

after doing what  @dbuezas said it did work but it seems to be a workaround and not the same behavior. As +- nicely explained here I think: https://youtu.be/X6nCKQToOUg more precise here: https://youtu.be/X6nCKQToOUg?si=2XWAhxUQf3ybh-2x&t=201

**Comment by Ardchie63** [comment]
The exclucive setting is very important for technical prints like screws or gears(heeringbone) without any test prints. It is one more needed step for designe, print, good result without playing.

**Comment by AJolly** [comment]
This is the feature I really want from cura.  Makes screws so much easier.

**Comment by p-dmitrij** [comment]
Very useful feature, it would be great to implement it!

**Comment by lxlolxl** [comment]
I agree that this feature is necessary and urgent, please slice a 3mm diameter hole parallel to the XY plane (perpendicular to the printing wall), you will find that there will be a section of path occupying the round hole at the top and bottom, but if you use cura's "exclude" slicing mode, you won't have this problem, and it's not something that can be solved by the XY expansion and hole expansion.

**Comment by arosovic** [comment]
> I'd suggest you do this:
> 
> 1. print an orca tolerance test
> 2. see where the printed pin fits as tight as you wish tolerances to be
> 3. Divide the number by 2
> 4. Set it as xy hole compensation
> 5. Print the tolerance test again and corroborate it now fits in the zero position but it is not too loose
> 
> And then just leave it like that for any print. That's what I like about this setting, it will look for holes and then correct for them by exactly that amount regardless of their size. I'm not sure but the value may differ for different line heights.

No, don't use hole compensation at all! (Use "Convert holes to polyholes" instead.)
It shoud be renamed to arc compensation, and reprogrammed accordingly because it has sense only for round holes, but it applies on square holes, too. It's sense should be to correct pulling filament towards the center of the hole, which is not present at square holes, but it is present at open arcs.

**Comment by FireflyJar** [comment]
I am also looking for this feature.

**Comment by Gismo6303** [comment]
Hello, I've switch for cura to orca today, and I need this feature too please

**Comment by BigRon1955** [comment]
This feature is critical for producing accurate threaded parts.  Not understanding why this is a feature in Cura, but Orca Slicer and Bambu Studio have both failed to recognize the significance of this much needed setting.  I really like Orca Slicer, and don't like to be forced to use Cura for threads!

**Comment by dbuezas** [comment]
Isn't it equivalent to "x-y hole compensation" and "x-y contour compensation"? Or rather, those are even more flexible.
I use those instead and don't have issues with threaded parts.

**Comment by BigRon1955** [comment]
No, it's not equivalent.  X-Y hole compensation is very similar to
scaling.  Slicing tolerance is determined by the intersecting angle of the
slice/surface, which means it will not be applied uniformly as x-y hole
compensation would be.  Refer to the detailed Cura explanation, and
comments from others, including YouTube videos illustrating the
difference.  As a CNC programmer, I understand the way these offsets are
applied to surfaces when hemstitching 3D surfaces with a ball nose endmill.

**Comment by rhiffsta** [comment]
this feature would be very useful. why there is still no activity after 1.5y from devs when several people show interest and also point out why it is useful/wanted.. could we please finally get a dev assigned to it or at least a statement of implementation plans?

**Comment by AJolly** [comment]
Agreed, it's the one feature cura has that I really miss

**Comment by antoniopegorelli** [comment]
It's a very nice feature for printers that are not as precise as Bambulab's models.
Even better if you want to print a downloaded STL and don't want to mess with scaling or redesigning to make it work.

**Comment by bespokecomp** [comment]
Wow.  You automatically closed a topic about a crucial feature that everyone wants!

Why!??

**Comment by Brad331** [comment]
OrcaSlicer, Bambu Slicer, Prusa Slicer, and Slic3r definitely do not handle tolerances on slopes correctly.
My part has a vertical hexagonal hole that is nominally 12.8mm wide. With hole compensation and contour compensation set appropriately, the vertical sides of the hexagon correctly measure 12.8mm, yet the sloped sides measure 12.59mm.
![Image](https://github.com/user-attachments/assets/031e517c-4233-4e0c-8723-03a0745cea13)
![Image](https://github.com/user-attachments/assets/0accbf41-a728-424f-a729-1a44d124fcd2)

**Comment by erentar** [comment]
Is there a way to prevent this issue from being auto-closed?

**Comment by BigRon1955** [comment]
I completely agree with you and others that the slicing tolerance option definitely needs to be implemented in Orca Slicer.  X-Y hole and contour offsets are only applied properly to surfaces which are perpendicular to the build plate.  A canted surface does not receive the proper offset.  Slicing tolerance is determined by the intersecting angle of the slice/surface, which means it will not be applied uniformly as x-y hole or contour compensation would be.  Refer to the detailed Cura explanation, and comments from others, including YouTube videos illustrating the difference.  As a CNC programmer, I understand the way these offsets are applied to non-perpendicular surfaces when hemstitching 3D surfaces with a ball nose endmill.  Laying down a _nearly_ round layer of filament is a very similar process.

**Comment by Ardchie63** [comment]
On Top I prefer a different solution. It is a good idea to have 2 Parameter normal, exclude, include and only XY direction vs all XYZ

**Comment by BigRon1955** [comment]
That would be ideal, to have both the normal, include and exclude options as well as retain the x-y compensation for holes and contours.  After doing some research, I found that the 2 methods of generating the offsets for g-code are in the way layers are sliced and how they intersect with the model geometry.

**Comment by Ardchie63** [comment]
The normal setting aktual now have some trouble with a bolt and a nut. The slicer reduced the backlash with the strategies for the angel of the thread. To get a funktional bold and nut you must change the Diameter, in the past I used Cura and you can build a thread with Fusion used this Parameter and get a Bold and nut that work fine.

**Comment by Ardchie63** [comment]
The bad thing for the implementation in Cura for exclude you lost one layer for Z. So I think it will be better to used the second parameter only XY to get better backlash and the real Z, vs the adjustment for XYZ like Cura.
